tests/integration_test/isomorphism_problem.py

```diff
@@ -1,29 +1,45 @@
 from algorithms.preprocessing import fix_degrees
 from tests.integration_test.algorithm_options import apply_could_be_isomorphic, apply_remove_twins, apply_tree_algorithm, branching_method, apply_complement
+from algorithms.color_initialization import degree_color_initialization
+
+
+def preprocessing(G: "Graph"):
+    """
+    In this method all pre-processing is done that need to be done only one time. This method is called before any
+    other computations are done.
+    :param G: The graph to be preprocessed
+    :return: Factor the amount of automorphisms need to be multiplied with to get the right result
+    """
+    G_complement = apply_complement(G)
+
+    fix_degrees(G)
```

Comment:
Is de fix degree nodig op beide grafen omdat je later ook nog de originele kunt gaan gebruiken?

Reply:
Omdat je het origineel altijd moet gebruiken voor het aantal automorphismen kleur ik die. En omdat je niet weet of G_complement echt het complement is of niet gebruik ik die gewoon altijd om te bepalen of grafen isomorph zijn. En verder is fix_degrees alleen een O(n) methode die een nieuwe property degree_fixed aan elke vertex geeft. Dus ik denk dat als het dubbel gaat dat niet heel veel uitmaakt voor de totale tijd.